sdk/core/azure-json-gson/src/main/java/com/azure/json/gson/GsonJsonProvider.java

```diff
@@ -17,6 +17,12 @@
  * Implementation of {@link JsonProvider} that creates instances using GSON.
  */
 public class GsonJsonProvider implements JsonProvider {
+    private static final String JSON_READER_EXCEPTION = "Both 'json' and 'options' must be passed as non-null to "
+        + "create an instance of JsonReader.";
+
+    private static final String JSON_WRITER_EXCEPTION = "Both 'json' and 'options' must be passed as non-null to "
+        + "create an instance of JsonWriter.";
```

Comment:
You require non-null here, but in the implementation you have code that allows for `options` to be null. Is there an inconsistency here?

Reply:
Yes, but the implementation can only be called by these APIs or other implementation code called by these APIs, meaning non-nullness should already be guaranteed. If it's a concern, I can add more null checks.